metricbeat/helper/metricset.go

```diff
@@ -125,8 +125,12 @@ func (m *MetricSet) createEvent(event common.MapStr, host string, rtt time.Durat
 	// Each metricset has a unique eventfieldname to prevent type conflicts
 	eventFieldName := m.Module.name + "-" + m.Name
 
+	// Selectors are applied first
 	event = applySelector(event, m.Config.Selectors)
```

Comment:
As discussed I suggest to remove the selectors in favor of filter:include_fields.